src/mono/mono/metadata/domain.c

```diff
@@ -128,6 +136,8 @@ mono_init_internal (const char *root_domain_name)
 
 	mono_counters_register ("Max HashTable Chain Length", MONO_COUNTER_INT|MONO_COUNTER_METADATA, &mono_g_hash_table_max_chain_length);
 
+	ensure_stack_size ();
+
```

Comment:
Why is this needed ? It doesn't look like something that would work on all platforms.

Reply:
This fixes https://github.com/dotnet/runtime/issues/76805, but indeed this is something that, like in coreclr, should be turned on with a `ENSURE_PRIMARY_STACK_SIZE` definition.

Reply:
Pushed conditional use of `ensure_stack_size`, modeled after coreclr.

Reply:
I'd suggest moving this to mini_init () in mini-runtime.c, thats where all the arch specific execution engine stuff is.